modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/util/TaskFilterExecutor.java

```diff
@@ -185,6 +187,30 @@ private boolean checkFilterForNumber(Filter filter, BigDecimal param) {
         return result;
     }
 
+    private boolean checkFilterForBoolean(Filter filter, Boolean param) {
+        OperatorType operatorType = OperatorType.fromString(filter.getOperator());
+        boolean result = false;
+        boolean expressionValue = Boolean.parseBoolean(filter.getExpression());
+
+        if (operatorType != null) {
+            switch (operatorType) {
+                case EXIST:
+                    result = true;
```

Comment:
this should we check if the param is true? (result = param) If it always returns true it's pointless